### PR TITLE
Return 403 when access is denied

### LINKED ISSUE #16: Return '403' Forbidden error message to an unauthorized user
**Current**
When user access is denied, an error code of 401 is thrown.
Unauthorized401

**Expected**
It should return _403_ Forbidden error message to the unauthorized user.

**It worked by:**
Updating exception to **_ForbiddenOperationException.class_** instead of throwing _AuthenticationException.class_ in
https://github.com/google/fhir-access-proxy/blob/3bf8a400e257f234f677154442f63570cd131307/server/src/main/java/com/google/fhir/proxy/BearerAuthorizationInterceptor.java#L238
 
cc : @bashir2

### PR DESCRIPTION
**Pull request by bashir2**
## Description of what I changed
Fixes #16 

## E2E test

TESTED:

Manually ran the proxy and tested that 403 is returned instead of 401, when access is denied.

## Checklist: I completed these to help reviewers :)

- [x] I have read and will follow the [review process](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/doc/review_process.md).
- [x] I am familiar with Google Style Guides for the language I have coded in.
- [x] My IDE is configured to follow the Google [**code styles**](https://google.github.io/styleguide/).
- [x] I have **added tests** to cover my changes. (If you refactored existing code that was well tested you do not have to add tests)
- [x] I ran `mvn clean package` right before creating this pull request and added all formatting changes to my commit.
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: server/src/test/java/com/google/fhir/proxy/BearerAuthorizationInterceptorTest.java [hunk 1]
```diff
@@ -112,6 +117,24 @@ private String generateKeyPairAndEncode() {
     return null;
   }
 
+  private BearerAuthorizationInterceptor createTestInstance(boolean returnedCheck)
```
Comment (omarismail94) [inline server/src/test/java/com/google/fhir/proxy/BearerAuthorizationInterceptorTest.java:120]: nit: the `returnedCheck` var name is not clear as to what it represents. How about renaming to `isAccessGranted` or something similar?

**Review by omarismail94 — APPROVED** [review]
One small nit

File: server/src/test/java/com/google/fhir/proxy/BearerAuthorizationInterceptorTest.java (hunk 1 above)
Comment (bashir2) [reply server/src/test/java/com/google/fhir/proxy/BearerAuthorizationInterceptorTest.java]: Done

**Review by bashir2 — COMMENTED** [review]
Thanks for the review.

**Pull request merged**